From d82877c3adeb53bbbcb50ff918f12e0db778f2ee Mon Sep 17 00:00:00 2001
From: Eli Collins <eli@cloudera.com>
Date: Sun, 11 Mar 2012 18:42:13 -0700
Subject: [PATCH 1136/1179] HDFS-930. Better error message for DATA_TRANSFER_VERSION mismatched.

Add some context information in the IOException during a version
mismatch to help debugging.

Reason: Bug
Author: Karthik K
Ref: CDH-731
---
 .../hadoop/hdfs/server/datanode/DataXceiver.java   |    4 +++-
 1 files changed, 3 insertions(+), 1 deletions(-)

diff --git a/src/hdfs/org/apache/hadoop/hdfs/server/datanode/DataXceiver.java b/src/hdfs/org/apache/hadoop/hdfs/server/datanode/DataXceiver.java
index 21d965b..5aece5f 100644
--- a/src/hdfs/org/apache/hadoop/hdfs/server/datanode/DataXceiver.java
+++ b/src/hdfs/org/apache/hadoop/hdfs/server/datanode/DataXceiver.java
@@ -123,7 +123,9 @@ class DataXceiver extends Thread implements Runnable, FSConstants {
           }
           short version = in.readShort();
           if ( version != DataTransferProtocol.DATA_TRANSFER_VERSION ) {
-            throw new IOException( "Version Mismatch" );
+            throw new IOException("Version Mismatch (Expected: " +
+               DataTransferProtocol.DATA_TRANSFER_VERSION  +
+               ", Received: " +  version + " )");
           }
 
           op = in.readByte();
-- 
1.7.0.4

